docs/proposals/migration/design-of-seamless-cluster-migration-scheme.md

```diff
@@ -0,0 +1,291 @@
+       - name: nginx
+         image: nginx:latest
+```
+
+#### Story 3
+
+Similarly, if multiple `Deployment` is defined in one `PropagationPolicy` , and users hope `Karmada` ignoring takeover the conflict `Deployment` by default, but forcing takeover individual specificed conflict `Deployment` :
+
+A feasible practice is to declare `conflictResolution: ignore` in the `PropagationPolicy` (or leave it blank), and annotate `work.karmada.io/conflict-resolution: overwrite` in the `ResourceTemplate`.
+
+
+
+### Notes/Constraints/Caveats (Optional)
+
+If the `conflictResolution` field of the `PropagationPolicy` and the `conflict-resolution` annotation of the `ResourceTemplate` are specified in the same time, constraints below should be followed:
+
+1. Priority effectiveness： `conflict-resolution` annotation in `ResourceTemplate`  **>** `conflictResolution` field in `PropagationPolicy`
+2. Results corresponding to different pair values：
+
+| PP \ RT             | not set   | ignore | overwrite | null or invalid |
+|---------------------|-----------|--------|-----------|-----------------|
+| **not set**         | ignore    | ignore | overwrite | ignore          |
+| **ignore**          | ignore    | ignore | overwrite | ignore          |
+| **overwrite**       | overwrite | ignore | overwrite | overwrite       |
+| **null or invalid** | ignore    | ignore | overwrite | ignore          |
+
+> PP refer to PropagationPolicy ，RT refer to ResourceTemplate 
+
+
+
+### Risks and Mitigations
+
+none
+
+
+
+## Design Details
+
+
+
+### API Modify
+
+1、add `string` field `conflictResolution` to `PropagationSpec`
+
+```go
+ type PropagationPolicy struct {
+     metav1.TypeMeta   `json:",inline"`
+     metav1.ObjectMeta `json:"metadata,omitempty"`
+ 
+     // Spec represents the desired behavior of PropagationPolicy.
+     // +required
+     Spec PropagationSpec `json:"spec"`
+ }
+ 
+ // PropagationSpec represents the desired behavior of PropagationPolicy.
+ type PropagationSpec struct {
+     // conflictResolution how to resolve workload when resource template has conflicts with member clusters
+     conflictResolution string `json:"conflictResolution,omitempty"`  // TODO add
+     ...
+ }
+```
+
+2、add `string` field `conflictResolution` to `ResourceBindingSpec`
+
+```go
+ type ResourceBinding struct {
+     metav1.TypeMeta   `json:",inline"`
+     metav1.ObjectMeta `json:"metadata,omitempty"`
+ 
+     // Spec represents the desired behavior.
+     Spec ResourceBindingSpec `json:"spec"`
+ 
+     // Status represents the most recently observed status of the ResourceBinding.
+     // +optional
+     Status ResourceBindingStatus `json:"status,omitempty"`
+ }
+ 
+ // ResourceBindingSpec represents the expectation of ResourceBinding.
+ type ResourceBindingSpec struct {
+     // conflictResolution how to resolve workload when resource template has conflicts with member clusters
+     conflictResolution string `json:"conflictResolution,omitempty"`    // TODO add
```

Comment:
I don't think it is necessary to modify the api of `Work` and `ResourceBinding`, because they are invisible to users, and they can be declared through annotations.

Reply:
Good question, admire you for seeing this question at a glance.

I had the same opinion as you at the begining, honestly, there is no need to do it from the implementation perspective, no that modify even makes code more clean, but three reason changed my opinion:

1. we not only want to just solve the problem in this scenario, but also solemnly recommend the ability and practice of seamless migration to users. Adding this field to CRDs including `ResourceBinding` and `Work ` can more clearly demonstrate this ability to users than adding annotations.
2. adding annotations is just a **compatible** way for individual exceptions, even if we remove it, it's still justifiable. Assuming it doesn't exist, we still need to modify the api of `Work` and `ResourceBinding`. I mean, the annotation is just a addons, our desgin shouldn't overdependence on it.

hope for your more viewpoint about this question~

Reply:
I think at least `ResourceBinding` doesn't need to be modified, because it can combine `PropagationPolicy`. but for `Work` API, I reserve my opinion and see if others think differently.